docs/design/storage-node-deployment-windows-home.md

```diff
@@ -0,0 +1,165 @@
+    * Must be run with admin privileges.
+    * Installs the automatic updater binary and error gui application. (msi)
+    * Sets up automatic updater as a windows Service with sufficient privileges.
+* Automatic Updater (binary)
+    * Downloads storage node binary, Sets up storage node, and runs the watchdog process.
+    * Doesn't start the storage node if storage node has not created a valid config.
+    * Send error reports to satellite.
+    * Writes update related errors to log file.
+    * Watchdog process
+        * Monitors storage node by periodically sending messages to pulse endpoint on storage node and waiting for responses.
+        * Restarts the storage node if a crash/unresponsiveness is detected.
+* Storage Node (binary)
+    * Shares drive with satellite network.
+    * Writes storage node operation related errors to log file
+    * Has pulse endpoint that can be hit for knowing if the storage node is alive and healthy. 
+* Error gui application
+    * Shows errors from log file
+    * Notifies user of service errors.
+    * Saves last reported error timestamp to a file for knowing if there are unread errors.
+
+### Starting Storage Node binary on OS Start-Up
+
+We need to ensure that updater binary starts on computer start-up,
+without logging into the system, and this updater binary launches the storage node This is achieved when installing via the msi.
+Avoid triggering UAC.
+
+### Restarting Storage Node binary on Crash / Problems
+
+We need to ensure that storage node binary restarts after a crash.
+
+* detect crashes and detect unresponsiveness
+    * updater binary checks pulse of storage node binary with ipc messages. storage node will have a pulse endpoint and the updater hits that endpoint with timeouts.
+    * A windows service can be configured to restart the service on fail/crash.
+   
+### Logging
+
+* Log to disk.
+* Rotate files.
+* Compress old stuff.
+* Delete really old stuff.
+* Ensure we limit the size of logs...
+
+### Resource Limits
+
+Ensure we:
+* can set limits to memory usage,
+* can set limits to CPU usage.
+* send graceful shutdown message and restart storage node if memory usage is too high.
+* Windows specific apis also exist for limiting process memory and cpu usage.
+* limit CPU might be to limit number of cores it runs on in go.
+
+## Testing
+
+Verify program can be run with windows defender firewall and at least one other 3rd party firewall.
+
+## Design Overview
+
+### Automatic Updates and Handling failures
+
+General
+* Windows firewall and other 3rd party firewalls can block storage node operations.
+    * [isportallowed](https://docs.microsoft.com/en-us/windows/win32/api/netfw/nf-netfw-inetfwmgr-isportallowed) windows api function can be used to make sure we are allowed through firewall
+    * Can we add code to detect if firewall is blocking storage node operations?
+    * Unblock storage node operator in firewall settings.
+    * Can we detect if windows firewall is running?
```

Comment:
Design should describe the design not ask questions. Move these into Open Questions section if they are unresolved.